### DISCUSSION TITLE
Can I dynamically register SLAAC IPv6 addresses with CoreDNS?

### DISCUSSION THREAD

**Discussion by pcgeek86** [coredns/coredns#5206] [Q&A] [answered]
I'd like to register SLAAC IPv6 addresses on my local network with a CoreDNS server. Is this possible, and if so, how could I go about setting up this type of environment? Just spin up CoreDNS, and point all my devices at that DNS server's IP address?

**Comment by chrisohaver** [comment, marked as answer]
No, CoreDNS doesn't support this.

Thinking on how it _could_ be supported via new feature, SLAAC nodes would need to actively register their hostnames to CoreDNS somehow, or CoreDNS would have to generate random names, or base the names on the IPv6 addresses (neither of which would be particularly useful).

**Comment by pcgeek86** [comment]
Yeah the devices (ie. Windows 11, MacOS, Linux) would have to register their network interfaces with CoreDNS. Should I put in a feature request to support SLAAC registration?

**Reply by chrisohaver** [reply]
You may. I think more generally it would fall under the category of adding support for dynamic updates (DDNS), which CoreDNS does not support.  IIUC, SLAAC requires a DHCPv6 server to do this registration, and then the DHCP server would do a DDNS update to the DNS server.  It's a very large task to add DDNS support, and unlikely to be implemented soon, but doesn't hurt to ask.
